### ISSUE TITLE
Add `defineConfig` function

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#8783]
For typescript configurations, it would be useful if a `defineConfig` function could be exported from `stylelint`, to be used like this in `stylelint.config.ts`:

```ts
import {defineConfig} from "stylelint";
export default defineConfig({/* config */});
```

The implementation of the function is a simple [identity function](https://en.wikipedia.org/wiki/Identity_function). The type definition then declares the first argument type as `Config`, which makes type checking of the passed configuration work.

Notably, this already exists as userland module in https://github.com/stylelint-types/stylelint-define-config, but I found that that module is inaccurate, so I prefer if this would be included in stylelint itself, which also guarantees that the validation is performed for the correct stylelint version.

**Comment by jeddy3** [comment]
@silverwind Thank you for the suggestion.

ESLint added a `defineConfig` function [in March](https://eslint.org/blog/2025/03/flat-config-extends-define-config-global-ignores/#main) to improve the experience of TypeScript users. It seems to be an emerging pattern in the configs of other tools, too.

ESLint also uses `defineConfig` to automatically flatten the arguments to eliminate some confusion around the use of the spread operator, and reintroduce the `extends` property. Neither of which applies to our current config format. Regardless, improving the experience for TypeScript users seems worthwhile.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by silverwind** [comment]
PR is up: https://github.com/stylelint/stylelint/pull/8824

**Comment by silverwind** [comment]
In case someone is looking for alternatives, this is already possible today:

```ts
import type {Config} from "stylelint";

export default {} satisfies Config;
```

**Comment by ybiquitous** [comment]
If we no longer need the new API, should we close this issue?

**Comment by silverwind** [comment]
From my point of view, using `satisfies Config` is clearly superior than having this function, so I'd vote for close. I can contribute a documentation update for it.

**Comment by ybiquitous** [comment]
@silverwind I'd really appreciate it if you could improve the documentation. Feel free to open a PR with the v17 branch as a base.

**Comment by niksy** [comment]
I still think it would be useful to have `defineConfig`, at least because as @jeddy3 said it’s a pattern that most of the tools are following.

**Comment by jeddy3** [comment]
Let's go with documenting `satisfies Config` for now, so that we have a single signature for defining configs as there's potential to confuse our users if we add another.

This may change in the future if, like ESLint did with needing to automatically flatten their arguments, we have another need to introduce `defineConfig`.

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/8908
